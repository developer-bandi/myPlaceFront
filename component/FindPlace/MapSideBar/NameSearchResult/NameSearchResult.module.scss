.mainBlock {
  overflow-y: scroll;
  height: calc(100vh - 315.25px);
  padding: 10px;
}

.ectStatus {
  text-align: center;
  font-size: 25px;
  margin-top: 40px;
  color: gray;
}

.loading {
  margin: 200px 100px 0 100px;
}
