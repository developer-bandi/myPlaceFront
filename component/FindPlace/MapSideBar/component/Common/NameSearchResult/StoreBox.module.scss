.storeinfo {
  padding: 20px;
  padding-bottom: 10px;
  border-bottom: 1px solid #e5e5e5;
  cursor: pointer;
  &:hover {
    opacity: 0.6;
  }
}

.titlebox {
  display: flex;
  align-items: center;
  margin-bottom: 10px;
}

.name {
  font-size: 20px;
  font-weight: bold;
}

.category {
  color: gray;
  margin-left: 10px;
}

.taglist {
  display: flex;
  white-space: nowrap;
  overflow-x: scroll;
  margin-top: 15px;
  padding-bottom: 10px;
}

.tag {
  margin: 0 15px 5px 0;
  font-size: 13px;
}
