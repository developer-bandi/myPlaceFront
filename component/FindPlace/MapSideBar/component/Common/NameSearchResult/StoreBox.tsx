import { SearchResultType } from "../../../../../../lib/apitype/search";
import styles from "./StoreBox.module.scss";

interface StoreBoxProps {
  searchResult: SearchResultType[];
  showStoreInfo: (storeId: number) => void;
}

const StoreBox = ({ searchResult, showStoreInfo }: StoreBoxProps) => {
  return (
    <>
      {searchResult.map((storeInfo) => {
        return (
          <div
            key={storeInfo.name}
            className={styles.storeinfo}
            onClick={() => {
              showStoreInfo(storeInfo.id);
            }}
          >
            <div className={styles.titlebox}>
              <h3 className={styles.name}>{storeInfo.name}</h3>
              <p className={styles.category}>{storeInfo.category}</p>
            </div>
            <p className={styles.adress}>{storeInfo.dist / 1000}km</p>
            <ul className={styles.taglist}>
              {Object.keys(storeInfo.hashtag)
                .sort(function (a, b) {
                  if (storeInfo.hashtag !== undefined) {
                    return storeInfo.hashtag.a - storeInfo.hashtag.b;
                  }
                  return 0;
                })
                .slice(0, 6)
                .map((hashtagname) => {
                  if (storeInfo.hashtag !== undefined)
                    return (
                      <li key={hashtagname} className={styles.tag}>
                        #{`${hashtagname}  ${storeInfo.hashtag[hashtagname]}`}
                      </li>
                    );
                })}
            </ul>
          </div>
        );
      })}
    </>
  );
};

export default StoreBox;
