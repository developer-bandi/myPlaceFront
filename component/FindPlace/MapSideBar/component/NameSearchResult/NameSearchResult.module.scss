.mainBlock {
  height: 655px;
  overflow-y: scroll;
  @media screen and (max-width: 1023px) {
    height: 296.75px;
  }
}

.ectStatus {
  text-align: center;
  font-size: 25px;
  margin-top: 40px;
  color: gray;
}

.loading {
  margin: 200px 100px 0 100px;
}
