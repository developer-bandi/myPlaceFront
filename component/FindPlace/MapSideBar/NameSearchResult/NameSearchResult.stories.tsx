import { ComponentStory, ComponentMeta } from "@storybook/react";
import NameSearchResult from "./NameSearchResut";
import { action } from "@storybook/addon-actions";

export default {
  title: "FindPlace/MapSideBar/NameSearchResult",
  component: NameSearchResult,
  parameters: {
    layout: "fullscreen",
  },
} as ComponentMeta<typeof NameSearchResult>;

const Template: ComponentStory<typeof NameSearchResult> = (arg) => {
  return (
    <div style={{ width: 340, position: "relative" }}>
      <NameSearchResult {...arg} showStoreInfo={action("showStoreInfo")} />
    </div>
  );
};

export const isLoading = Template.bind({});
isLoading.args = {
  searchResult: { loading: true, error: false },
};

export const isError = Template.bind({});
isError.args = {
  searchResult: { loading: false, error: true },
};

export const isBeforeSearch = Template.bind({});
isBeforeSearch.args = {
  searchResult: { loading: false, error: false },
};

export const isNoResult = Template.bind({});
isNoResult.args = {
  searchResult: { content: [], loading: false, error: false },
};

export const isNormal = Template.bind({});
isNormal.args = {
  searchResult: {
    content: [
      {
        id: 1,
        name: "상점",
        category: "카페",
        latitude: "test",
        longitude: "test",
        dist: 10,
        hashtag: {
          아름다운: 5,
          깨끗한: 4,
        },
      },
    ],
    loading: false,
    error: false,
  },
};
