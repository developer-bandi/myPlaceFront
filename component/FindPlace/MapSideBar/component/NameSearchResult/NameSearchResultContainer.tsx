import {useDispatch, useSelector} from "react-redux";
import {RootReducer} from "../../../../../store";
import {getStoreInfo} from "../../../../../store/reducers/storeInfo/Reducer";
import NameSearchResult from "./NameSearchResut";

const NameSearchResultContainer = () => {
  const searchResult = useSelector((state: RootReducer) => state.searchResult);
  const dispatch = useDispatch();
  const showStoreInfo = (storeId: number) => {
    dispatch(getStoreInfo(storeId));
  };

  return (
    <NameSearchResult
      searchResult={searchResult}
      showStoreInfo={showStoreInfo}
    />
  );
};

export default NameSearchResultContainer;
