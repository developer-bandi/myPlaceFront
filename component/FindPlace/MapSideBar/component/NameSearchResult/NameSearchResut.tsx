import Image from "next/image";
import { SearchResultState } from "../../../../../store/reducers/searchResult/searchResultReducer";
import searchResultLoading from "../../../../../public/searchResultLoading.gif";
import StoreBox from "../Common/NameSearchResult/StoreBox";
import styles from "./NameSearchResult.module.scss";

interface NameSearchResult {
  searchResult: SearchResultState;
  showStoreInfo: (storeId: number) => void;
}

const NameSearchResult = ({
  searchResult,
  showStoreInfo,
}: NameSearchResult) => {
  return (
    <div className={styles.mainBlock}>
      {searchResult.loading ? (
        <div className={styles.loading}>
          <Image src={searchResultLoading} alt="loading"></Image>
        </div>
      ) : searchResult.error !== null ? (
        <p className={styles.ectStatus}>에러가 발생하였습니다</p>
      ) : searchResult.content.length === 0 ? (
        <p className={styles.ectStatus}>검색결과가 없습니다</p>
      ) : (
        <StoreBox
          searchResult={searchResult.content}
          showStoreInfo={showStoreInfo}
        />
      )}
    </div>
  );
};

export default NameSearchResult;
