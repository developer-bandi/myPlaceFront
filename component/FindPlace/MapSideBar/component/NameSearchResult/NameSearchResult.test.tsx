import {render, screen} from "@testing-library/react";
import NameSearchResult from "./NameSearchResut";

describe("NameSearchResult Presentational 테스트", () => {
  it("로딩중", () => {
    const utils = render(
      <NameSearchResult
        searchResult={{loading: true, error: false}}
        showStoreInfo={jest.fn()}
      ></NameSearchResult>
    );
    expect(utils.container).toMatchSnapshot();
    screen.getByTestId("loading");
  });
  it("에러발생", () => {
    const utils = render(
      <NameSearchResult
        searchResult={{loading: false, error: true}}
        showStoreInfo={jest.fn()}
      ></NameSearchResult>
    );
    expect(utils.container).toMatchSnapshot();
    screen.getByTestId("error");
  });
  it("컨텐츠 없음", () => {
    const utils = render(
      <NameSearchResult
        searchResult={{loading: false, error: false}}
        showStoreInfo={jest.fn()}
      ></NameSearchResult>
    );
    expect(utils.container).toMatchSnapshot();
    screen.getByTestId("noResult");
  });
  it("컨텐츠 있음", () => {
    const utils = render(
      <NameSearchResult
        searchResult={{
          content: [
            {
              id: 1,
              name: "testName",
              category: "testCategory",
              latitude: "testLatitude",
              longitude: "testLongitude",
              dist: 1,
              hashtag: {
                testHashtag: 1,
              },
            },
          ],
          loading: false,
          error: false,
        }}
        showStoreInfo={jest.fn()}
      ></NameSearchResult>
    );
    expect(utils.container).toMatchSnapshot();
    screen.getByTestId("result");
  });
});
