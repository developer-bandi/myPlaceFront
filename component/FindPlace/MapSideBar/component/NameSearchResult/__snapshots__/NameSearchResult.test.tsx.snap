// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NameSearchResult Presentational 테스트 검색하기전 1`] = `
<div>
  <div
    class="mainBlock"
    data-testid="start"
  >
    <p
      class="ectStatus"
    >
      원하는 장소를 검색해보세요
    </p>
  </div>
</div>
`;

exports[`NameSearchResult Presentational 테스트 로딩중 1`] = `
<div>
  <div
    class="mainBlock"
    data-testid="loading"
  >
    <div
      class="loading"
    >
      <span
        style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
      >
        <span
          style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
        >
          <img
            alt=""
            aria-hidden="true"
            src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2724%27%20height=%2724%27/%3e"
            style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
          />
        </span>
        <img
          alt="loading"
          data-nimg="intrinsic"
          decoding="async"
          src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
          style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
        />
        <noscript />
      </span>
    </div>
  </div>
</div>
`;

exports[`NameSearchResult Presentational 테스트 에러발생 1`] = `
<div>
  <div
    class="mainBlock"
    data-testid="error"
  >
    <p
      class="ectStatus"
    >
      에러가 발생하였습니다
    </p>
  </div>
</div>
`;

exports[`NameSearchResult Presentational 테스트 컨텐츠 없음 1`] = `
<div>
  <div
    class="mainBlock"
    data-testid="noResult"
  >
    <p
      class="ectStatus"
    >
      검색결과가 없습니다
    </p>
  </div>
</div>
`;

exports[`NameSearchResult Presentational 테스트 컨텐츠 있음 1`] = `
<div>
  <div
    class="mainBlock"
    data-testid="result"
  >
    <div
      class="storeinfo"
    >
      <div
        class="titlebox"
      >
        <h3
          class="name"
        >
          testName
        </h3>
        <p
          class="category"
        >
          testCategory
        </p>
      </div>
      <p
        class="adress"
      >
        0.001
        km
      </p>
      <ul
        class="taglist"
      >
        <li
          class="tag"
        >
          #
          testHashtag  1
        </li>
      </ul>
    </div>
  </div>
</div>
`;
